projects/ngx-maplibre-gl/src/lib/source/video-source.component.ts

```diff
@@ -64,7 +66,6 @@ export class VideoSourceComponent
       urls: this.urls(),
       coordinates: this.coordinates(),
     };
-    this.mapService.addSource(this.id(), source);
-    this.sourceId.set(this.id());
+    this.sourceDirective.addSource(source);
```

Comment:
no need to set the id?

Reply:
Source directive holds the id and uses it.